coderd/files/cache.go

```diff
@@ -43,15 +98,30 @@ type Cache struct {
 	lock sync.Mutex
 	data map[uuid.UUID]*cacheEntry
 	fetcher
+
+	// metrics
+	currentOpenFileReferences prometheus.Gauge
+	totalOpenFileReferences   prometheus.Counter
+
+	currentOpenFiles prometheus.Gauge
+	totalOpenedFiles prometheus.Counter
+
+	currentCacheSize prometheus.Gauge
+	totalCacheSize   prometheus.Counter
```

Comment:
this is complicated enough that it should probably be a separate struct that's just a member of the actual `Cache`